strix/utilities/click_callbacks.py

```diff
@@ -446,10 +447,106 @@ def dump_params(ctx, param, value, output_path=None, skip_flag=True):
                 if (not p.prompt or p.is_flag or p.prompt_cond) and p.name in dump_dict
             ]:
                 dump_dict.pop(flag)
-            json.dump(dump_dict, f, indent=2, sort_keys=True, cls=PathlibEncoder)
+            yaml.dump(dump_dict, f, sort_keys=True)
     return value
 
 
+def confirmation(
+    ctx,
+    param,
+    value,
+    output_dir: Optional[str] = None,
+    output_dir_ctx: Optional[str] = None,
+    save_code_dir: Optional[str] = None,
+    exist_ok: Optional[bool] = False,
+    checklist: Optional[Union[Callable, Sequence[Callable]]] = None,
+):
+    """Callback for confirmation
+
+    You can use functools.partial() to modify the params
+    e.g. functools.partial(confirmation, output_dir='./')
```

Comment:
This function is used in CLI or python files？I think maybe it's not convenient to modify the params in this way...

Reply:
Used in python (as click's callback function). 
See https://github.com/Project-Strix/Strix/blob/master/strix/main_entry.py#L140-L148